library/promos.ts

```diff
@@ -283,3 +267,53 @@ export const getPromoByContentId = async (contentId: string): Promise<any> => {
 
   return promo;
 };
+
+export const getTotalPromoEarnedByUser = async (
+  userId: string,
+  promoId: string
+): Promise<number> => {
+  const userTotalMsatEarned = await db
+    .knex("promo_reward")
+    .join("promo", "promo_reward.promo_id", "promo.id")
+    .where({ "promo.id": promoId, "promo_reward.user_id": userId })
+    .andWhere("promo_reward.is_pending", false)
+    .sum(
+      db.knex.raw(
+        "promo.msat_payout_amount * promo_reward.msat_amount as total_msat_earned"
+      )
+    )
```

Comment:
Need to fix this part of the query
```suggestion
    .sum("promo.msat_payout_amount as total_msat_earned")
    .groupBy("promo.msat_payout_amount")
```